File: quicksort/quicksort.js

var array = [5, 9, 3, 6, 11];

function quickSort(left, right) {

    if (left > right) {
        return;
    }

    var i = left, j = right, temp = array[left];
    var t;

    while (i < j) {

        while (i < j && array[j] >= temp) {
            j--;
        }

        while (i < j && array[i] <= temp) {
            i++;
        }

        if (i < j) {
            t = array[i];
            array[i] = array[j];
            array[j] = t;
        }
    }



    t = array[left];
    array[left] = array[i];
    array[i] = t;

    quickSort(left, i-1);
    quickSort(i+1, right);
}

quickSort(0, array.length-1);
console.log(array);

File: QuickSort/quicksort.go
package main

import (
	"fmt"
)

 var a = make([]int, 11)
// var a [101] int

func quickSort(left, right int) {
	var i, j, temp int
	if left > right {
		return
	}

	temp = a[left]
	i, j = left, right
	for i != j {
		for a[j] >= temp && i < j {
			j--
		}
		for a[i] <= temp && i < j {
			i++
		}

		if i < j {
			a[i], a[j] = a[j], a[i]
		}
	}

	a[left], a[i] = a[i], a[left]

	quickSort(left, i-1)
	quickSort(i+1, right)
}

func main() {
	var n int
	fmt.Scan(&n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%d", &a[i])
	}

	quickSort(0, n-1)
	for i := 0; i < n; i++ {
		fmt.Printf("%d ", a[i])
	}
}
